### PR TITLE
fix: installing solo when dependencies are not installed globally

### LINKED ISSUE #3137: bug: kubectl dependency check passes but later kubectl spawn fails (ENOENT) in one-shot
### Describe the bug

 solo one-shot single deploy reports:

✔ Check dependency: kubectl ...

…but later fails with:

container call failed to start: spawn kubectl ENOENT

This means the dependency check is not equivalent to the runtime execution environment, or the check is not actually validating presence/executability in the same way the runtime code path needs.

### Describe the expected behavior

If kubectl will be invoked later via child_process.spawn("kubectl", ...), the dependency check must guarantee that will succeed (or record and use an absolute path to kubectl).

If kubectl is missing, fail fast before long-running deploy steps.

### To Reproduce

Remove kubectl from the system (you did sudo apt remove --purge kubectl).

Install Solo via brew (brew install hiero-ledger/tools/solo).

Run solo one-shot single deploy.

Observe: dependency check passes, deploy proceeds to consensus network, then fails at solo consensus node setup with spawn kubectl ENOENT.

### Additional Context

Failure occurs during solo consensus node setup → Fetch platform software into network nodes → Update node: node1.

Stack traces:

default-one-shot.ts:525

k8-client-container.ts:64

This is painful because it fails after the solo consensus network deploy [6m19s].

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
## Description

This PR fixes several issues resulting from missing dependencies. 

- There were several instances where the `kubectl` command was called in a hard-coded way. This caused an `ENOENT` error if it was not found in the PATH environment variable. This was fixed by passing a full path to the executable file, located in the `~/.solo/bin` directory.

- The latest release of `vfkit` uses an updated asset name format, which resulted in an error during the automatic dependency download procedure. This was fixed by specifying the exact release version to be downloaded for the following dependencies: `vfkit`, `podman`, `gvproxy`, and by improving the asset name matching.

- Improved the automatic kind cluster creation using Podman. Solo no longer attempts to start a podman machine if it already exists.

### Related Issues

* Closes #3137
* Related to #

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: src/integration/kube/k8-client/k8-client-factory.ts [hunk 1]
```diff
@@ -3,11 +3,22 @@
...
 
 @injectable()
 export class K8ClientFactory implements K8Factory {
   private readonly k8Clients: Map<string, K8> = new Map<string, K8>();
+  private readonly kubeExecutable: string;
+
+  public constructor(@inject(InjectTokens.KubectlInstallationDir) installationDirectory: string) {
+    this.kubeExecutable = path.join(
+      patchInject(installationDirectory, InjectTokens.KubectlInstallationDir, K8ClientFactory.name),
+      'kubectl',
```
[the remark this answers is not in the document]
Comment (Ivo-Yankov) [reply src/integration/kube/k8-client/k8-client-factory.ts]: I added the constant to constants.ts since it will also be used in other places.
